web/src/routes/Event.jsx

```diff
@@ -8,19 +13,76 @@ import { Table, Thead, Tbody, Th, Tr, Td } from '../components/Table';
 export default function Event({ eventId }) {
   const apiHost = useApiHost();
   const { data, status } = useEvent(eventId);
+  const [showDialog, setShowDialog] = useState(false);
+  const [deleteStatus, setDeleteStatus] = useState(FetchStatus.NONE);
+  const [deleteMessage, setDeleteMessage] = useState();
+
+  const handleClickDelete = () => {
+    setShowDialog(true);
+  };
+
+  const handleDismissDeleteDialog = () => {
+    setShowDialog(false);
+  };
+
+
+  const handleClickDeleteDialog = useCallback(async () => {
+
+    setDeleteStatus(FetchStatus.LOADING);
+    let success;
+    try {
+      const response = await fetch(`${apiHost}/api/events/${eventId}`, { method: 'DELETE' });
+      const deleteEvent = await getJSON(response)
+      success = deleteEvent.success;
```

Comment:
You can simplify this with destructuring:
```suggestion
      const { success = false, message } = await getJSON(response)
```